ballista/core/src/async_reader/mod.rs

```diff
@@ -376,6 +133,8 @@ impl<R: AsyncBufRead + Unpin + Send> AsyncStreamReader<R> {
         &mut self,
     ) -> BoxFuture<'_, Result<Option<RecordBatch>, ArrowError>> {
         if self.finished {
+            BALLISTA_ASYNC_STREAM_READER_LATENCY
```

Comment:
I would add this in a `Drop` implementation instead of here

Reply:
Nice